File: app/src/main/java/com/example/admin/smarttranslator/Fragments/HistoryFragment.java
package com.example.admin.smarttranslator.Fragments;

import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.support.v7.widget.GridLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.example.admin.smarttranslator.Adapters.HistoryAdapter;
import com.example.admin.smarttranslator.Models.User;
import com.example.admin.smarttranslator.R;

public class HistoryFragment extends Fragment {

    public static HistoryFragment newInstance() {
        return new HistoryFragment();
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        return inflater.inflate(R.layout.history_fragment, container, false);
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {

        RecyclerView recyclerView = getView().findViewById(R.id.historyView);
        recyclerView.setHasFixedSize(true);
        RecyclerView.LayoutManager layoutManager = new GridLayoutManager(getView().getContext(), 2);
        recyclerView.setLayoutManager(layoutManager);

        if(User.getPhotoCardStorage().size()==0){
//            photos.add(new PhotoCard("en", "ru"));
        } else{
            RecyclerView.Adapter adapter = new HistoryAdapter(User.getPhotoCardStorage(), getView().getContext());
            recyclerView.setAdapter(adapter);

        }


    }

}

File: app/src/main/java/com/example/admin/smart_translator/Fragments/HistoryFragment.kt
package com.example.admin.smart_translator.Fragments

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup

import com.example.admin.smart_translator.Adapters.HistoryAdapter
import com.example.admin.smart_translator.Models.User
import com.example.admin.smart_translator.R

class HistoryFragment : androidx.fragment.app.Fragment() {

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return inflater.inflate(R.layout.history_fragment, container, false)
    }

    override fun onViewCreated(@NonNull view: View, @Nullable savedInstanceState: Bundle?) {

        val recyclerView = getView()!!.findViewById<androidx.recyclerview.widget.RecyclerView>(R.id.historyView)
        recyclerView.setHasFixedSize(true)
        val layoutManager = androidx.recyclerview.widget.GridLayoutManager(getView()!!.context, 2)
        recyclerView.layoutManager = layoutManager

        if (User.photoCardStorage.size == 0) {
            //            photos.add(new PhotoCard("en", "ru"));
        } else {
            val adapter = HistoryAdapter(User.photoCardStorage, getView()!!.context)
            recyclerView.adapter = adapter

        }


    }

    companion object {

        fun newInstance(): HistoryFragment {
            return HistoryFragment()
        }
    }

}
